packages/upswyng-web/src/components/__tests__/Resource.test.tsx

```diff
@@ -1,17 +1,24 @@
 import React from "react";
 import { Resource } from "../Resource";
-import { foodResource } from "../../DataMocks";
-import { shallow } from "enzyme";
+import { mount } from "enzyme";
 import useResource from "../useResource";
 
 jest.mock("../Categories", () => ({
   categories: {},
 }));
-jest.mock("../../utils/searchParams");
-jest.mock("../../constants");
-jest.mock("../LoadingSpinner.tsx", () => "LoadingSpinner");
+jest.mock("../LoadingSpinner.tsx", () => () => "TEST-loading-spinner");
```

Comment:
While the original mock works for [shallow rending](https://airbnb.io/enzyme/docs/api/shallow.html), it wouldn't work for when [mounting](https://airbnb.io/enzyme/docs/api/ReactWrapper/mount.html).

I believe that in the former, loading spinner would be rendered as a normal HTML element during mount. The warnings you were seeing was the test complaining about a random `<LoadingSpinner/>` element that was being read as an HTML element rather than a component.

The updated mock changes the render of the loading spinner to be plain text. I also prepended the text with "TEST" to signify that this render is unique to this test and not actually rendered within the application.